src/main/java/com/optimaize/langdetect/LanguageDetectorImpl.java

```diff
@@ -199,11 +196,11 @@
     private double[] initProbability() {
         double[] prob = new double[ngramFrequencyData.getLanguageList().size()];
         if (priorMap != null) {
-            //TODO analyze and optimize this code, looks like double copy.
             System.arraycopy(priorMap, 0, prob, 0, prob.length);
-            for(int i=0;i<prob.length;++i) prob[i] = priorMap[i];
```

Comment:
The lines 203 and 204 do exactly the same job ie copying the priorMap array in the prob array. I don't think there just should be any update to the Javadoc for search a change.